// angular-admin-lte/esm2015/lib/box-small/box-small.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, Input, ContentChild } from '@angular/core';
import { BoxSmallContentDirective, BoxSmallFooterDirective, BoxSmallHeaderDirective } from './box-small.directive';
export class BoxSmallComponent {
    constructor() {
        this.contentStyleClass = 'small-box-content';
        this.footerStyleClass = 'small-box-footer';
        this.headerStyleClass = 'small-box-header';
        this.iconStyleClass = 'ion ion-bag';
        this.styleClass = 'small-box';
    }
}
BoxSmallComponent.decorators = [
    { type: Component, args: [{
                selector: 'mk-box-small',
                template: `<div [ngClass]="styleClass" [mkColor]="backgroundColor" mkColorProperty="background-color">
  <div class="inner">
    <h3 *ngIf="header || boxSmallHeaderDirective" [ngClass]="headerStyleClass" [mkFontColor]="headerColor">
      {{header}}
      <ng-content select="mk-box-small-header"></ng-content>
    </h3>
    <p [ngClass]="contentStyleClass" [mkFontColor]="contentColor">
      <ng-container *ngIf="boxSmallHeaderDirective || boxSmallContentDirective || boxSmallFooterDirective; else noDirective">
        <ng-content select="mk-box-small-content"></ng-content>
      </ng-container>
      <ng-template #noDirective>
        <ng-content></ng-content>
      </ng-template>
    </p>
  </div>
  <div *ngIf="iconStyleClass" class="icon">
    <i [ngClass]="iconStyleClass" [mkFontColor]="iconColor"></i>
  </div>
  <span *ngIf="footer || boxSmallFooterDirective" [ngClass]="footerStyleClass" [mkFontColor]="footerColor">
    {{footer}}
    <ng-content select="mk-box-small-footer"></ng-content>
  </span>
</div>
`,
                styles: [`.small-box.bg-color{color:#fff}/deep/ .small-box-footer:hover{cursor:pointer}/deep/ .small-box-footer a{color:rgba(255,255,255,.8)}/deep/ .small-box-footer:hover a{color:#fff}`]
            },] },
];
BoxSmallComponent.propDecorators = {
    backgroundColor: [{ type: Input }],
    contentColor: [{ type: Input }],
    contentStyleClass: [{ type: Input }],
    footer: [{ type: Input }],
    footerColor: [{ type: Input }],
    footerStyleClass: [{ type: Input }],
    header: [{ type: Input }],
    headerColor: [{ type: Input }],
    headerStyleClass: [{ type: Input }],
    iconColor: [{ type: Input }],
    iconStyleClass: [{ type: Input }],
    styleClass: [{ type: Input }],
    boxSmallHeaderDirective: [{ type: ContentChild, args: [BoxSmallHeaderDirective,] }],
    boxSmallFooterDirective: [{ type: ContentChild, args: [BoxSmallFooterDirective,] }],
    boxSmallContentDirective: [{ type: ContentChild, args: [BoxSmallContentDirective,] }]
};
if (false) {
    /** @type {?} */
    BoxSmallComponent.prototype.backgroundColor;
    /** @type {?} */
    BoxSmallComponent.prototype.contentColor;
    /** @type {?} */
    BoxSmallComponent.prototype.contentStyleClass;
    /** @type {?} */
    BoxSmallComponent.prototype.footer;
    /** @type {?} */
    BoxSmallComponent.prototype.footerColor;
    /** @type {?} */
    BoxSmallComponent.prototype.footerStyleClass;
    /** @type {?} */
    BoxSmallComponent.prototype.header;
    /** @type {?} */
    BoxSmallComponent.prototype.headerColor;
    /** @type {?} */
    BoxSmallComponent.prototype.headerStyleClass;
    /** @type {?} */
    BoxSmallComponent.prototype.iconColor;
    /** @type {?} */
    BoxSmallComponent.prototype.iconStyleClass;
    /** @type {?} */
    BoxSmallComponent.prototype.styleClass;
    /** @type {?} */
    BoxSmallComponent.prototype.boxSmallHeaderDirective;
    /** @type {?} */
    BoxSmallComponent.prototype.boxSmallFooterDirective;
    /** @type {?} */
    BoxSmallComponent.prototype.boxSmallContentDirective;
}
